plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -437,8 +437,17 @@ WITH PerfCounters AS (
 				THEN d.[name]
 			ELSE d.[physical_database_name]
 		END
-	WHERE
-		counter_name IN (
+	WHERE 
+		spi.instance_name NOT IN (
```

Comment:
There is a couple of problems here:
1. The instance_name column is nullable. If there is a NULL in a row, the results from NOT IN () may be unexpected.
2. We should also check object_name, because the same instance_name for a different object could carry a different meaning.

Both of these can be addressed with the following rewrite:

```sql
        -- filter out unnecessary SQL DB system database counters, other than master and tempdb
        NOT (spi.object_name LIKE 'MSSQL%:Databases%' AND spi.instance_name IN ('model','model_masterdb','model_userdb','msdb','mssqlsystemresource'))
        AND
        (
		counter_name IN (
...
```

Reply:
Ah got it, I've updated the PR with the changes you suggested. Thanks!